app/com/arpnetworking/rollups/RollupGenerator.java

```diff
@@ -73,9 +74,10 @@
      * For each metric name received, perform the following:
      *
      *     1. Retrieve the tag names for this metric (TagNamesMessage)
-     *     2. Query for the last data point given the maximum backfill and set of tags (LastDataPointMessage)
-     *     3. If this datapoint is in the past, generate a backfill job for the period
-     *     furthest in the past, and enqueue it by sending to the RollupManager. (FinishRollupMessage)
+     *     2. For each period size:
+     *         2a. Query for the last data point given the maximum backfill and set of tags (LastDataPointMessage)
+     *         2b. If this datapoint is in the past, generate a backfill job for the period
+     *         furthest in the past, and enqueue it by sending to the RollupManager. (FinishRollupMessage)
```

Comment:
Nothing right now, the assumption is there will never be mixed data since a second job would just overwrite the same datapoints.

That's all captured in a separate ticket.